FlowCrypt/Controllers/Msg/MessageViewController.swift

```diff
@@ -144,35 +147,36 @@ final class MessageViewController: TableNodeViewController {
 extension MessageViewController {
     private func fetchDecryptAndRenderMsg() {
         showSpinner("loading_title".localized, isUserInteractionEnabled: true)
-
-        Promise { [weak self] in
-            guard let self = self else { return }
-            let promise = self.messageService.getAndProcessMessage(
-                with: self.input.objMessage,
-                folder: self.input.path
-            )
-            let message = try awaitPromise(promise)
-            self.processedMessage = message
-        }
-        .then(on: .main) { [weak self] in
-            self?.handleReceivedMessage()
-        }
-        .catch(on: .main) { [weak self] error in
-            self?.handleError(error)
+        Task {
+            do {
+                let rawMimeData = try awaitPromise(self.messageProvider.fetchMsg(message: self.input.objMessage, folder: self.input.path))
```

Comment:
I think this will run on main thread

Reply:
Will fetching message in async way block the main thread? In other words, is it acceptable to do that on main thread or should we be using other thread, and if so, how?